gcc/rust/backend/rust-compile-resolve-path.cc

```diff
@@ -35,15 +35,29 @@ namespace Compile {
 void
 ResolvePathRef::visit (HIR::QualifiedPathInExpression &expr)
 {
-  resolved = resolve (expr.get_final_segment ().get_segment (),
-		      expr.get_mappings (), expr.get_locus (), true);
+  auto final_segment = HIR::PathIdentSegment::create_error ();
+  if (expr.is_lang_item ())
+    final_segment
+      = HIR::PathIdentSegment (LangItem::ToString (expr.get_lang_item ()));
```

Comment:
Could this end up getting confused if there are multiple `Some` declarations floating around? 

I.e. 

fn Some () {
 let _ = option_env!("PWD"); // Yields lang-item path to Some("/tmp")
}

Reply:
Maybe - I'm adding a different implementation right now which is more precise and makes use of the lang item's known NodeId

Reply:
I think the version you have up now works, at least from tracing the code by eye.